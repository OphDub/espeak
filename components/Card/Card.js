import React, { Component } from 'react';
import { TouchableOpacity, StyleSheet, Text, View, TextInput } from 'react-native';
import AwesomeAlert from 'react-native-awesome-alerts';

type Props = {};
export default class Card extends Component {
  constructor() {
    super();
    this.state = {
      answer: '',
      showAlert: false,
      showHint: false,
    }
  }

  showAlert = () => {
    this.setState({ showAlert: true })
  }

  hideAlert =() => {
    this.setState({ showAlert: false })
  }

  showHint = () => {
    this.setState({ showHint: true });
  }

  hideHint = () => {
    this.setState({ showHint: false })
  }

  handleChange = (text) => {
    this.setState({answer: text})
  }
  
  handleHideHint = () => {
    this.props.onHintPoints();    
    this.hideHint();
  }

  handleSubmit = () => {
    const { answer } = this.state;
    const { onCorrectAnswer, word } = this.props;
    const { spanish } = word;
    if (answer.toLowerCase() === spanish.toLowerCase()) {
      onCorrectAnswer(word);
    } else {
      this.showAlert();
    }

    this.setState({ answer: '' });
  }

  render() {
    const { english, spanish, hint } = this.props.word;
    const { showAlert, showHint } = this.state;

    return (
      <View style={styles.container}>
        <Text style={styles.question}>How do you say...</Text>
        <View style={styles.engCard}>
        {
        this.props &&
          <Text style={styles.engText}>{ english.toUpperCase() }</Text>
        }
        </View>
        <Text style={styles.question}>in Spanish?</Text>
        <TextInput 
          placeholder='Type word in Spanish'
          onChangeText={(text) => this.handleChange(text)}
          value={this.state.answer}
          style={styles.spanInput}
        />
        <TouchableOpacity
          onPress={() => this.showHint()}
          style={styles.submitBtn}
        >
          <Text style={styles.btnText}>Need a hint?</Text>
        </TouchableOpacity> 
        <TouchableOpacity
          onPress={() => this.handleSubmit()}
          style={styles.submitBtn}
        >
          <Text style={styles.btnText}>SUBMIT</Text>
        </TouchableOpacity> 
        <AwesomeAlert
          show={showHint}
          showProgress={false}
          title="Here's a hint"
          message={hint}
          closeOnTouchOutside={false}
          closeOnHardwareBackPress={false}
          showCancelButton={false}
          showConfirmButton={true}
          confirmText="OK, got it!"
          confirmButtonColor="#3AAFb9"
          onConfirmPressed={() => {
            this.handleHideHint();
          }}
        />
        <AwesomeAlert
          show={showAlert}
          showProgress={false}
          title="Almost..."
          message="You've got this!"
          closeOnTouchOutside={true}
          closeOnHardwareBackPress={false}
          showCancelButton={false}
          showConfirmButton={true}
          cancelText="No, cancel"
          confirmText="Try again!"
          confirmButtonColor="#3AAFb9"
          onCancelPressed={() => {
            this.hideAlert();
          }}
          onConfirmPressed={() => {
            this.hideAlert();
          }}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  btnText: {
    color: '#FFF',
  },
  question: {
    fontSize: 20,
  },
  engCard: {
    alignItems: 'center',
    borderColor: '#1E3888',
    borderRadius: 8,
    borderWidth: 2,
    backgroundColor: 'white',
    paddingVertical: 35,
    width: 220,
    marginVertical: 20,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  engText: {
    color: '#1E3888',
    fontSize: 25,
  },
  spanInput: {
    fontSize: 20,
    margin: 15,
    borderBottomWidth: 2,
    borderColor: '#3AAFB9',
    padding: 5,
    width: 250,
  },
  submitBtn: {
    backgroundColor: '#3AAFB9',
    padding: 10,
    borderRadius: 8,
    marginTop: 10,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  }
});
