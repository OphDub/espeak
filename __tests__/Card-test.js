import React from 'react';
import { shallow, configure } from 'enzyme';
import Card from '../components/Card/Card';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const word = { english: 'hi', spanish: 'hola', hint: 'o-la' };
  const wrap = renderer.create(<Card word={word}/>).toJSON();
  expect(wrap).toMatchSnapshot();
})