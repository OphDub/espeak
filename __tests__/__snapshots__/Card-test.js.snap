// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    How do you say...
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#1E3888",
        "borderRadius": 8,
        "borderWidth": 2,
        "marginVertical": 20,
        "paddingVertical": 35,
        "shadowColor": "#979797",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.7,
        "shadowRadius": 4,
        "width": 220,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1E3888",
          "fontSize": 25,
        }
      }
    >
      HI
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    in Spanish?
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Type word in Spanish"
    style={
      Object {
        "borderBottomWidth": 2,
        "borderColor": "#3AAFB9",
        "fontSize": 20,
        "margin": 15,
        "padding": 5,
        "width": 250,
      }
    }
    value=""
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3AAFB9",
        "borderRadius": 8,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#979797",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.7,
        "shadowRadius": 4,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
        }
      }
    >
      SUBMIT
    </Text>
  </View>
</View>
`;
